### PR TITLE
Sortededges

### PR DESCRIPTION
**Pull request by rutgervandenberg**

### CODE REVIEW

File: src/main/java/nl/tudelft/lifetiles/graph/models/jgrapht/JGraphTGraphAdapter.java [hunk 1]
```diff
@@ -182,4 +190,17 @@ private DefaultEdge unpackEdge(final Edge<V> input) {
         return ed.getInternalEdge();
     }
 
+    /**
+     * @author Rutger van den Berg
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/graph/models/jgrapht/JGraphTGraphAdapter.java:233]: Method description should be above the `@author` tag.

File: src/main/java/nl/tudelft/lifetiles/graph/models/jgrapht/JGraphTGraphAdapter.java [hunk 2]
```diff
@@ -182,4 +190,17 @@ private DefaultEdge unpackEdge(final Edge<V> input) {
...
+     *         Compares two edges by their target vertex.
+     */
+    class EdgeComparatorByVertex implements Comparator<Edge<V>> {
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/graph/models/jgrapht/JGraphTGraphAdapter.java:236]: This class is redundant, you could rewrite these 10 lines to a one-liner by using a lambda. See http://www.dreamsyssoft.com/java-8-lambda-tutorial/comparator-tutorial.php for examples.

File: src/test/java/nl/tudelft/lifetiles/graph/models/jgrapht/GraphAdapterTest.java [hunk 3]
```diff
@@ -33,8 +33,9 @@ public static void runOnce() {
     @Before
     public void setUp() throws Exception {
         gf = fp.getFactory("JGraphT");
-        v1 = new SequenceSegment(null, 0, 0, null);
-        v2 = new SequenceSegment(null, 0, 0, null);
+        v1 = new SequenceSegment(null, 1, 5, null);
+        v2 = new SequenceSegment(null, 6, 10, null);
+        v3 = new SequenceSegment(null, 7, 10, null);
         gr = gf.getGraph();
```
Comment (jorenham) [inline src/test/java/nl/tudelft/lifetiles/graph/models/jgrapht/GraphAdapterTest.java:43]: The graph should be loaded from the test data.

**Comment by albert-smit** [comment]
looks good if you fix the mentioned issues, and the sequenceSegment equals method

File: src/test/java/nl/tudelft/lifetiles/graph/models/jgrapht/GraphAdapterTest.java (hunk 3 above)
Comment (joswinter) [inline src/test/java/nl/tudelft/lifetiles/graph/models/jgrapht/GraphAdapterTest.java:43]: I disagree, we should just test the JGraphTGraphAdapter in this test without adding additional dependencies on the parser.

**Comment by joswinter** [comment]
Everything looks good.

**Comment by alangerak** [comment]
Looks good

File: src/main/java/nl/tudelft/lifetiles/graph/models/Graph.java.orig [hunk 4]
```diff
@@ -0,0 +1,105 @@
+package nl.tudelft.lifetiles.graph.models;
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/graph/models/Graph.java.orig:1]: Don't push this...

File: src/main/java/nl/tudelft/lifetiles/graph/models/sequence/SequenceSegment.java.orig [hunk 5]
```diff
@@ -0,0 +1,165 @@
+package nl.tudelft.lifetiles.graph.models.sequence;
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/graph/models/sequence/SequenceSegment.java.orig:1]: Uhhh...

**Comment by jorenham** [comment]
- You should remove the `.orig` files
- Don't create a new list for all sets, this causes unnecessary memory overhead, use SortedSet instead (i.e. http://tutorials.jenkov.com/java-collections/sortedset.html)

**Pull request closed without merging**